duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -172,302 +199,461 @@ namespace caf {
-      sr.LepNuAngle = truth[i].GetNeutrino().Nu().Momentum().Vect().Angle(truth[i].GetNeutrino().Lepton().Momentum().Vect());
-
-      sr.nP     = 0;
-      sr.nN     = 0;
-      sr.nipip  = 0;
-      sr.nipim  = 0;
-      sr.nipi0  = 0;
-      sr.nikp   = 0;
-      sr.nikm   = 0;
-      sr.nik0   = 0;
-      sr.niem   = 0;
-      sr.niother = 0;
-      sr.nNucleus = 0;
-      sr.nUNKNOWN = 0;
-
-      sr.eP = 0.;
-      sr.eN = 0.;
-      sr.ePip = 0.;
-      sr.ePim = 0.;
-      sr.ePi0 = 0.;
-      sr.eOther = 0.;
-
-      for( int p = 0; p < truth[i].NParticles(); p++ ) {
-        if( truth[i].GetParticle(p).StatusCode() == genie::kIStHadronInTheNucleus ) {
-
-          int pdg = truth[i].GetParticle(p).PdgCode();
-          double ke = truth[i].GetParticle(p).E() - truth[i].GetParticle(p).Mass();
-          if     ( pdg == genie::kPdgProton ) {
-            sr.nP++;
-            sr.eP += ke;
-          } else if( pdg == genie::kPdgNeutron ) {
-            sr.nN++;
-            sr.eN += ke;
-          } else if( pdg == genie::kPdgPiP ) {
-            sr.nipip++;
-            sr.ePip += ke;
-          } else if( pdg == genie::kPdgPiM ) {
-            sr.nipim++;
-            sr.ePim += ke;
-          } else if( pdg == genie::kPdgPi0 ) {
-            sr.nipi0++;
-            sr.ePi0 += ke;
-          } else if( pdg == genie::kPdgKP ) {
-            sr.nikp++;
-            sr.eOther += ke;
-          } else if( pdg == genie::kPdgKM ) {
-            sr.nikm++;
-            sr.eOther += ke;
-          } else if( pdg == genie::kPdgK0 || pdg == genie::kPdgAntiK0 || pdg == genie::kPdgK0L || pdg == genie::kPdgK0S ) {
-            sr.nik0++;
-            sr.eOther += ke;
-          } else if( pdg == genie::kPdgGamma ) {
-            sr.niem++;
-            sr.eOther += ke;
-          } else if( genie::pdg::IsHadron(pdg) ) {
-            sr.niother++; // charm mesons, strange and charm baryons, antibaryons, etc.
-            sr.eOther += ke;
-          } else if( genie::pdg::IsIon(pdg) ) {
-            sr.nNucleus++;
-          } else {
-            sr.nUNKNOWN++;
-          }
+  //------------------------------------------------------------------------------
 
-        }
-      }
+  void CAFMaker::FillBeamInfo(caf::SRBeamBranch &beam, const art::Event &evt) const
+  {
+    //TODO: See how to retrieve these info
+    beam.ismc = true; //Hardcoded to true at the moment.
+  }
 
-      // Reweighting variables
+  //------------------------------------------------------------------------------
 
-      // Consider
-      //systtools::ScrubUnityEventResponses(er);
+  void CAFMaker::FillCVNInfo(caf::SRCVNScoreBranch &cvnBranch, const art::Event &evt) const
+  {
+    art::Handle<std::vector<cvn::Result>> cvnin = evt.getHandle<std::vector<cvn::Result>>(fCVNLabel);
+
+    if( !cvnin.failedToGet() && !cvnin->empty()) {
+      cvnBranch.isnubar = (*cvnin)[0].GetIsAntineutrinoProbability() > 0.5; //Hardcoded 0.5 as threshold for nu/nubar
```

Comment:
Instead you could just record the probability so that the analyser can decide on the cut value.  There shouldn't be a hardcoded cut value here.

Reply:
I agree, but this is for now implemented as a bool. I'll ask Jeremy to change this for the future.

Reply:
Are the CVN values also stored?

Reply:
All the other values are stored in the normal way. This is the only field for which there is only a bool.

Reply:
great. thanks